FIREFLY-30: Add support for proprietary data display in table

loitly opened the pull request:
https://jira.ipac.caltech.edu/browse/FIREFLY-30

Adds ability to handle table with proprietary data.
- introduced DATARIGHTS_COL and RELEASE_DATE_COL column metas.
  - When one of these columns exists, table contains proprietary data.
  - has access is when DATARIGHTS_COL resolved to `true` or RELEASE_DATE_COL is less than today's date
- renders table row with pink when user does not have access
- added warning to DownloadDialog when a table with proprietary data is selected for download
- added additional JS unit tests to` TableUtil-Test.js`
- added Firefly API test: #7-Test Proprietary Data Display

Also:
Added Links to META-INFO API
 - You can inject `links` into a column via API's META-INFO.
 - added Firefly API test: 6 - Column links via API
 - you can also use ./firefly/demo/ffapi-table-test.html to test this.

Added `expand/close` feature to Firefly API test.
- converted a couple of table's components into functional components.

firefly test: https://irsawebdev9.ipac.caltech.edu/FIREFLY-30/firefly/
firefly API test: https://irsawebdev9.ipac.caltech.edu/FIREFLY-30/firefly/test/

ejoliet commented on src/firefly/js/tables/TableUtil.js:
```diff
@@ -936,6 +937,47 @@ export function getColsByType(tblColumns=[], type=COL_TYPE.ALL) {
+    return rcname || dcname ? [rcname, dcname] : [];
+}
+
+export function hasRowAccess(tableModel, rowIdx) {
```
@deannaji This is Loi's nice little util function that can be called from your side to govern the preview tab layout.

loitly commented on src/firefly/html/test/styles.css:
```diff
@@ -137,3 +143,17 @@ ul.expected-list li{
+    position: fixed;
+    border: 2px solid #b3b3b3;
+    height: calc(100% - 50px) !important;
+    width: calc(100% - 50px) !important;
```
This is used for `expand`.  The logic is to add this style to the div, then remove it when `close` is clicked.  Without `!important`, the `height` in the style attribute of the div will take precedence and the height here is ignored.  That's why I use `!important` to make is overwrite everything.

loitly commented:
@tgoldina , I'll add the check to TAP search data's tab to show the error message laid out in item 4.  BTW, I really liked the use of TAP search to test this PR.  It demonstrates the intent of this PR's implementation.  Thanks for thoughtful review.

loitly commented:
@tgoldina , I updated https://irsawebdev9.ipac.caltech.edu/FIREFLY-30/firefly/firefly-dev.html with the latest fix.  Please check and approve PR if it addressed your concerns.

@robyww , will you take a quick look at my latest commit and see if that's where you would have put the `has access` logic?  Thanks.
